File: paint.m
% 'o'	Circle
% '+'	Plus sign
% '*'	Asterisk
% '.'	Point
% 'x'	Cross
% '_'	Horizontal line
% '|'	Vertical line
% 'square' or 's'	Square
% 'diamond' or 'd'	Diamond
% '^'	Upward-pointing triangle
% 'v'	Downward-pointing triangle
% '>'	Right-pointing triangle
% '<'	Left-pointing triangle
% 'pentagram' or 'p'	Five-pointed star (pentagram)
% 'hexagram' or 'h'	Six-pointed star (hexagram)

x = 0:2:20;
% null
l.p1 = [0.05480471,0.01082922,0.00099901,0.00007284,0.00002404,0.00001643,0.00001268,0.00001082,0.00001085,0.00001047,0.00001023];
l.p2 = [0.17831602,0.11262871,0.05583773,0.01456559,0.00277051,0.00130143,0.00101074,0.00085550,0.00076331,0.00072396,0.00071003];
l.p3 = [0.24855752,0.19477913,0.13313952,0.07083177,0.02912350,0.01370124,0.00970833,0.00848493,0.00788818,0.00750563,0.00725166];
% null
l.p4 = [0.02171068,0.00637318,0.00102656,0.00015885,0.00004707,0.00002949,0.00002663,0.00002487,0.00002480,0.00002578,0.00002585];
l.p5 = [0.12515495,0.08952220,0.04703724,0.01852669,0.00743698,0.00442422,0.00366771,0.00352422,0.00354948,0.00361042,0.00364043];
l.p6 = [0.19864225,0.17553503,0.14429460,0.10658034,0.06967096,0.04314577,0.02913255,0.02326751,0.02115013,0.02042448,0.02018047];


f=figure;
f.Position = [0 42 1050 840];
LineWidth = 2;
MarkerSize = 18;
p1=semilogy(x,l.p1,'b-o','LineWidth',LineWidth,'MarkerSize',MarkerSize);grid on; hold on
p2=semilogy(x,l.p2,'b-s','LineWidth',LineWidth,'MarkerSize',MarkerSize);grid on; hold on
p3=semilogy(x,l.p3,'b-^','LineWidth',LineWidth,'MarkerSize',MarkerSize);grid on; hold on
p4=semilogy(x,l.p4,'r--o','LineWidth',LineWidth,'MarkerSize',MarkerSize);grid on; hold on
p5=semilogy(x,l.p5,'r--s','LineWidth',LineWidth,'MarkerSize',MarkerSize);grid on; hold on
p6=semilogy(x,l.p6,'r--^','LineWidth',LineWidth,'MarkerSize',MarkerSize);grid on; hold on

set(gca,'XTick',x,'FontName','Times','fontsize',18)
% title({'i.i.d. AER'},'FontWeight','normal','fontsize',24, 'interpreter', 'latex')
xlabel('SNR (dB)','FontName','Times','fontsize',24, 'interpreter', 'latex')
ylabel('Null','FontName','Times','fontsize',24, 'interpreter', 'latex')
axis([x(1) x(end) 1e-5 .4]);
grid on
gcf_set = [p1 p4 p2 p5 p3 p6];
lgd = legend(gcf_set, {'proposed UE=5', 'Compare \enspace UE=5', 'proposed UE=10', 'Compare \enspace UE=10', 'proposed UE=15', 'Compare \enspace UE=15'});
lgd.Interpreter = 'latex'; lgd.Location = 'northeast'; lgd.FontSize = 14;

File: paint.py
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import ticker


test_SNR = np.arange(0, 22, 2)
AER = np.abs(np.random.randn(3, 11)/1e3)
xtick = np.arange(0, 21, 5)
fontsmall = 12
fontbig = 15
width = .8

rc = {"font.family" : "serif", "mathtext.fontset" : "stix"}
plt.rcParams.update(rc)
plt.rcParams["font.serif"] = ["Times New Roman"] + plt.rcParams["font.serif"]
fig, ax = plt.subplots()
fig.set_size_inches(6.3, 5.25)
kw = dict(markersize=11.3, markerfacecolor='none', markeredgewidth=1, linewidth=.8, clip_on=False)    # clip: outside marker
ax.semilogy(test_SNR, AER[0], 'b-o', label=r'Proposed $\mathit{k}$ = 5' , **kw)
ax.semilogy(test_SNR, AER[1], 'r-->', label=r'Proposed $\mathit{k}$ = 10', **kw)
ax.semilogy(test_SNR, AER[2], 'k:s', label=r'Proposed $\mathit{k}$ = 15', **kw)
legend = ax.legend(loc='lower left', fontsize=fontsmall, markerscale=.8, framealpha=1, edgecolor='black', fancybox=False, borderpad=.25)
legend.get_frame().set_linewidth(width)
ax.set_xlabel('SNR (dB)', fontsize=fontbig)
ax.set_ylabel('ADEP', fontsize=fontbig)
ax.grid(which='major', alpha=.6)
ax.grid(which='minor', linestyle = ':', alpha=.5)
ax.set_xticks(xtick)
ax.set_xlim(test_SNR[0], test_SNR[-1])
ax.set_yscale('log',nonpositive='mask')
ax.tick_params(axis="x", which='both', direction="in", pad=6, labelsize=fontsmall)    # pad: distance to border
ax.tick_params(axis="y", which='both', direction="in", pad=6, labelsize=fontsmall)
# ax.yaxis.set_minor_formatter(ticker.ScalarFormatter())    # scalar not power
# ax.yaxis.set_minor_formatter(ticker.FormatStrFormatter('%.1f'))    # float number
# ax.yaxis.set_major_formatter(ticker.ScalarFormatter())
# ax.yaxis.set_major_formatter(ticker.FormatStrFormatter('%.1f'))

# box and arrow
ymax = np.min(AER)
xmax = np.argmin(AER[np.where(AER==ymax)[0][0]])
text = "min: {:.1f}e-2\n{:.1f}e-2\n{:.1f}e-4".format(min(AER[2])*1e2, min(AER[1])*1e2, min(AER[0])*1e4)
xytext = (0.53,0.155)    # box location
bbox_props = dict(boxstyle="square,pad=0.3", fc="w", ec="k", lw=width)    # pad: box size
arrowprops=dict(arrowstyle="->")
kw = dict(xycoords='data',textcoords="axes fraction",
            bbox=bbox_props, ha="right", va="top", arrowprops=arrowprops)
ax.annotate(text, xy=(test_SNR[xmax], ymax), xytext=xytext, fontsize=fontsmall, **kw)    # xy: arrow destination

# upper left scale
ax.text(-0.002, 1.02, r'$\times 10^{3}$',    # xy loc, text
        fontsize=fontsmall,
        horizontalalignment='center',
        verticalalignment='center',
        rotation=0,
        transform=ax.transAxes)
plt.savefig('snr.pdf')
plt.show()
